modules/dashboard/php/ajaxcharthelper.class.inc

```diff
@@ -0,0 +1,286 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used with c3.js for the Dashboard.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\dashboard;
+
+use Database;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Tara Campbell <tara.campbell@mail.mcgill.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxChartHelper extends \NDB_Page
+{
+    public $skipTemplate = true;
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        /**
+         * Ensure it's a POST request.
+         */
+        if ($request->getMethod() != 'POST') {
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(405)
+                ->withHeader('Allow', 'POST')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Only POST requests are allowed'
+                    )
+                );
+        }
+        /**
+         * Parse POST request body.
+         */
+        $requestBody = $request->getParsedBody();
+
+        /**
+         * Response to reply.
+         */
+        $response = $this->processRequest($requestBody);
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus(200)
+            ->withHeader('Allow', 'POST')
+            ->withBody(
+                new \LORIS\Http\StringStream(
+                    json_encode($response)
+                )
+            );
+    }
+    /**
+     * Processes the values and saves to database
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function processRequest(array $values) : array
+    {
+        $response = array();
+        if (isset($values['command'])
```

Comment:
instead of multiple if-elseif and checking each time if isset($values['command']), check it once and assign a null value if not set. `$command = $values['command'] ?? null;`
Then, a switch control structure would be appropriate and cleaner in my opinion.